test/unit/settings_test.py

```diff
@@ -145,6 +145,26 @@ def test_eia860_years_overlap_eia860m_years(self: Self):
                 eia860m_year_months=[bad_eia860m_year_month],
             )
 
+    def test_eia860m_years_overlap_eia860m_years(self: Self):
+        """Test validation error is raised when eia860m years overlap."""
+        max_eia860_year = max(Eia860Settings().years)
+        acceptable_eia860m_year = max_eia860_year + 1
+        bad_eia860m_year_months = [
+            f"{acceptable_eia860m_year}-01",
+            f"{acceptable_eia860m_year}-05",
+        ]
```

Comment:
I think there will be times when these inputs fail for the wrong reason, not because they overlap, but because there's no EIA-860M data available for May yet in the `max(eia860.years) + 1` year.  If you use Jan and Feb I think you won't have that problem, since if the new "early release" data for EIA-860 for 2023 comes out in June, there *should* be Jan and Feb 2024 data available by then.